ceph_deploy/hosts/remotes.py

```diff
@@ -179,10 +187,11 @@ def listdir(path):
     return os.listdir(path)
 
 
-def makedir(path, ignored=None):
+def makedir(path, ignored=None, uid=-1, gid=-1):
     ignored = ignored or []
     try:
         os.makedirs(path)
+        os.chown(path, uid, gid);
```

Comment:
os.chown() is not recursive. But it is not easy to implement it recursive (I tried and it has some problems with remote specification, locally it is just a few lines). But is is enough that the last dir has proper access.
(Moreover, I did not find any use that it actually creates more than actual dir. Maybe the whole makedirs is overkill there.)